src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
By the way, when you create your own `sync::SyncHistory` objects, which you do when you create `SharedGroup` objects, then it is up to you whether you want to pass a cooker to it, but it makes little sense to do that, because it will never be used. This is because the cooker is only invoked during integration of changesets received from the server, and those are always processed internally in the history objects created internally by the `sync::Session` objects.